de.metas.vertical.pharma/src/main/java/de/metas/vertical/pharma/PharmaOrderLineQuickInputValidator.java

```diff
@@ -0,0 +1,88 @@
+package de.metas.vertical.pharma;
+
+import java.util.Collections;
+
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.util.Services;
+import org.compiere.util.Env;
+import org.compiere.util.Util;
+import org.springframework.stereotype.Component;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.i18n.IMsgBL;
+import de.metas.i18n.ITranslatableString;
+import de.metas.i18n.TranslatableStringBuilder;
+import de.metas.order.IOrderLineQuickInputValidator;
+import de.metas.product.ProductId;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * metasfresh-pharma
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@Component
```

Comment:
why `@Component` as opposed to `@Service`?

Reply:
no particular reason. just that it's not a `@Service`

Reply:
as just talked: we are both not really sure..so imho let's keep it a `@Component` if you prefer